File: javascript/6-kyu/backspaces-in-string/solution.js
function cleanString(s) {
  let answer = [];
  for (let i = 0; i < s.length; i++) {
    if (s[i] === "#") {
      answer.pop();
    } else {
      answer.push(s[i]);
    }
  }
  return answer.join().replace(/,/g, "");
}

File: csharp/6-kyu/backspaces-in-string/solution.cs
using System;
using System.Text;

public class Kata
{
  public static string CleanString(string s)
  {
    var answer = new StringBuilder(s.Length);
    for (var i = 0; i < s.Length; i++)
    {
      if (!s[i].Equals('#'))
      {
        answer.Append(s[i]);
      }
      else if (s[i].Equals('#') && s.Length > 0)
      {
        if (answer.Length > 0)
          answer.Remove(answer.Length - 1, 1);
      }
    }
    return answer.ToString();
  }
}